### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by blessing-michael**
In this project we;
-  Built these interfaces:
   - The home page.
   - The comments popup.
- followed the layout of the wireframes provided. 
- Added a functionalities for when the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- When the user clicks on the "Comments" button, the Comments popup appears.
- Home page header and navigation similar to the given mockup.
- Comments popup
    - The selected API and shows details about the selected item.
       - The selected API and shows details about the selected item.
 - When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

### CODE REVIEW

File: src/modules/movies-count.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    const response = await axios.get('https://api.tvmaze.com/shows/1/episodes');
+    const result = response.data;
+    const movies = result.splice(0, 12);
+    movieCount.innerHTML = movies.length.toString();
+  } catch (error) {
+    console.error('Error:', error);
+  }
+};
+
+export default count;
```
Comment (mckent05) [inline src/modules/movies-count.js:16]: - Good job done so far, but you have not correctly implemented the movie test counter, according to the project requirements, `A counter function should look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page`. Please implement your counter to count the number of movies displayed, similar to what you did for the `comment` counter. Please fix this 🙏

File: test/move-count.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  it('test_api_call_failure', async () => {
+    const mockAxios = jest.spyOn(axios, 'get').mockRejectedValue(new Error('API call failed'));
+    const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
+    await count();
+    expect(mockAxios).toHaveBeenCalledWith('https://api.tvmaze.com/shows/1/episodes');
+    expect(consoleSpy).toHaveBeenCalledWith('Error:', new Error('API call failed'));
+  });
+
+  it('test_api_response_data_not_in_expected_format', async () => {
```
Comment (mckent05) [inline test/move-count.test.js:27]: - You have done well so far, but I have an error when I try running your tests, Please fix this 🙏 
t5t34t5

- Also after implementing a proper counter for the movie count, please add the correct tests 😄

File: README.md [hunk 3]
```diff
@@ -1 +1,149 @@
...
+This webapp is inspired by the movies from https://github.com/izebiafe/Blefi/issues.
+We would like to express our heartfelt gratitude to Microverse for providing us with this incredible project opportunity.
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) license
+d.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (mckent05) [inline README.md:149]: - You did well so far, but your readme seems muddled up on my PC, please confirm from your end that it displays properly 🙏 
wefwef

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @blessing-michael, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job working on your JavaScript project 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (Izebiafe) [reply README.md:149]: @mckent05 thank you so much for your review.
But every thing seem to work really well from my end and my README also work properly.
go over the code again and check if you might have missed something. thank you again for the review.

cc @blessing-michael

Comment (mckent05) [reply README.md:149]: - Okay @Izebiafe  I suggest you work on the first required change and then you can submit it for a review, another reviewer will pick it up and approve it, if required. Thank you

File: src/modules/movies-count.js (hunk 1 above)
Comment (danmainah) [reply src/modules/movies-count.js:16]: - [ ] @Team Great work implementing the changes requested, however as previously requested kindly consider changing how you have implemented the movies count method, it's supposed to use DOM elements as you did for the comments counter function in commentCount.js file.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @blessing-michael and @mckent05 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented the change requested to ensure the readme is formatted correctly.
✔️  Implemented the change requested to ensure all tests are running successfully.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**